### PR TITLE
Model uuid format with Guid in CSharp, UUID in Java

### PR DESCRIPTION
**Pull request by jhancock93**
Addresses issue #481. format="uuid" is modeled as Guid in CSharp and with UUID class in Java. Added 3 acceptance tests for multiple languages to check that the format is properly modeled and can be serialized/deserialized.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@jhancock93__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by jhancock93** [comment]
I needed to regenerate a couple java test fixtures after rebasing my code on the latest master branch.

**Comment by amarzavery** [comment]
Node.js supports uuid via package ["node-uuid"](https://github.com/broofa/node-uuid).
One can generate a uuid if required and have a guid regex to validate whether a given string is a valid guid by doing this:

```javascript
var uuid = require('node-uuid');
var sample_guid = uuid.v4();
if (sample_guid.match(/^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$/ig) === null) {
 throw Error (util.format('The provided uuid \'%s\' is invalid');
}
```

File: AutoRest/Generators/Java/Java.Tests/src/test/java/fixtures/bodyarray/ArrayTests.java [hunk 1]
```diff
@@ -1,558 +1,583 @@
...
+    @Test
+    public void putUuidValid() throws Exception {
+        client.getArrayOperations().putUuidValid(Arrays.asList(UUID.fromString("6dcc7237-45fe-45c4-8a6b-3a8a3f625652"),
+        UUID.fromString("d1399005-30f7-40d6-8da6-dd7c89ad34db"), UUID.fromString("f42f6aa1-a5bc-4ddf-907e-5f915de43205")));
+    }
+
+    @Test
+    public void getUuidInvalidChars() throws Exception {
+        try {
+            List<UUID> result = client.getArrayOperations().getUuidInvalidChars().getBody();
```
Comment (jianghaolu) [inline AutoRest/Generators/Java/Java.Tests/src/test/java/fixtures/bodyarray/ArrayTests.java]: Please add `fail()` after this line.

Comment (jhancock93) [inline AutoRest/Generators/Java/Java.Tests/src/test/java/fixtures/bodyarray/ArrayTests.java]: Thanks. Added Assert.fail() as suggested.

**Pull request merged**